subxt/src/config/extrinsic_params.rs

```diff
@@ -27,7 +27,7 @@ pub trait ExtrinsicParams<Index, Hash>: Debug + 'static {
     fn new(
         spec_version: u32,
         tx_version: u32,
-        nonce: Index,
+        nonce: impl Into<u64>,
```

Comment:
I'd keep this simple and just have it take a concrete `u64`; it's only called internally anyway, and so let's avoid extra generics unless they make the API more ergonomic for users :)